ide/gototest/src/org/netbeans/api/gototest/TestOppositesLocator.java

```diff
@@ -0,0 +1,194 @@
+            if (locator.appliesTo(fo)) {
+                if (locator.asynchronous()) {
+                    locator.findOpposite(fo, caretOffset, new LocationListener() {
+                        @Override
+                        public void foundLocation(FileObject fo, TestLocator.LocationResult location) {
+                            if (location != null) {
+                                FileObject fileObject = location.getFileObject();
+                                if(fileObject == null) {
+                                    String msg = location.getErrorMessage();
+                                    if (msg != null) {
+                                        DialogDisplayer.getDefault().notify(
+                                                new NotifyDescriptor.Message(msg, NotifyDescriptor.INFORMATION_MESSAGE));
+                                    }
+                                } else {
+                                    locationResults.put(location, fileObject.getName());
+                                }
+                            }
+                            allDone.countDown();
+                        }
+                    });
+                } else {
+                    TestLocator.LocationResult opposite = locator.findOpposite(fo, caretOffset);
+
+                    if (opposite != null) {
+                        FileObject fileObject = opposite.getFileObject();
+                        if (fileObject == null) {
+                            String msg = opposite.getErrorMessage();
+                            if (msg != null) {
+                                DialogDisplayer.getDefault().notify(
+                                        new NotifyDescriptor.Message(msg, NotifyDescriptor.INFORMATION_MESSAGE));
+                            }
+                        } else {
+                            locationResults.put(opposite, fileObject.getName());
+                        }
+                    }
+
+                    allDone.countDown();
+                }
+            }
+        }
+
+        try {
+            allDone.await();
+        } catch (InterruptedException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+
+        return locationResults.entrySet()
+                              .stream()
+                              .map(e -> new NamedLocation(e.getKey(),
+                                                          e.getValue()))
+                              .collect(Collectors.toList());
+    }
+
+    private TestLocator getLocatorFor(FileObject fo) {
+        Collection<? extends TestLocator> locators = Lookup.getDefault().lookupAll(TestLocator.class);
+        for (TestLocator locator : locators) {
+            if (locator.appliesTo(fo)) {
+                return locator;
+            }
+        }
+
+        return null;
+    }
+
+    private TestLocator.FileType getFileType(FileObject fo) {
+        TestLocator locator = getLocatorFor(fo);
```

Comment:
This is a little strange: while `populateLocationResults`  considers all locators. this implementation accepts the first result.